Sending generated ships to backend

RafalJDev opened the pull request:
Can you please review our project

RafalJDev committed:
Fist working commit (GET mapping are ok, POST mappings still not work)

RafalJDev committed:
Fist working commit (GET mapping are ok, POST mappings still not work)

Commit:
adding package json

Commit:
adding angular 5.1.3 core

RafalJDev committed:
First Board !!!!!!!!!

RafalJDev committed:
Merge remote-tracking branch 'origin/angular_attemps' into temp-brach

RafalJDev committed:
Merged two branches

RafalJDev committed:
Merged two branches

RafalJDev committed:
Change .gitignore

RafalJDev committed:
Change .gitignore

RafalJDev committed:
Soooooo I have done a lot of mini-experiments that are worth to push

RafalJDev committed:
Fixing missing front-ship service

RafalJDev committed:
Img directory removed

RafalJDev committed:
Separating backend and frontend

RafalJDev committed:
Merge remote-tracking branch 'origin/tryingMoveFrontendToAnotherModule' into tryingMoveFrontendToAnotherModule

# Conflicts:
#	backend/src/main/resources/static/assets/ship.png

Commit:
Refactor of pom.xml in modules. Added Dockerfile in backend

RafalJDev committed:
Added routing to front and basic configuration screen

Commit:
Node installation has been fixed

RafalJDev committed:
Added very basic ship dragging

RafalJDev committed:
First attempts to drap double mast ship

RafalJDev committed:
Merge remote-tracking branch 'origin/configuring_modularized_project' into tryingMoveFrontendToAnotherModule

# Conflicts:
#	frontend/src/assets/ship_1.png
#	pom.xml

RafalJDev committed:
Merging this branch with devops branch

RafalJDev committed:
Changing hardcoded host url, to retrieved one

RafalJDev committed:
Testing asynchronous request with angular

RafalJDev committed:
Added generating random ships on frontend on horizontal (pure typescript)

RafalJDev committed:
Added random ship placement on vertical, but it is working 70-80% time,
because of not proper boundary somewhere

RafalJDev committed:
Frontend changes:
Ok so random is now working correctly for full fleet, on both directions.
How it works ?
First its generating y, x indexes for board (0-9), it checks if
there is no ship on surroundings and if so it place the ship on board.

There few small bugs in code, that broke whole.

I made directories refactoring. Now there are 3 main directories models,
services, views.

All multi classes that were in one class are separated
to corresponding directories.

RafalJDev committed:
Finishing previous refactoring

RafalJDev committed:
Pushing cause I got to go home. T.B.C.

RafalJDev committed:
Refactoring & cleaning. Adeed proper ship sending to backend

Commit:
fix minnor bug in forntend title and remove unnecessary resources

RafalJDev committed:
Basic refactoring of packages in backend, deleted sout, comments,
changed names of components.

Nothing special.

gacikjakub commented:
Build
13
Successful
Running instance of this is on:
tryingFixSpringInDocker.battleships.kuba005.nazwa.pl

tomeks86 commented on backend/src/main/java/pl/krkteam/config/WebSecurityConfig.java:
```diff
@@ -0,0 +1,29 @@
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.cors();
+    http.csrf().disable();
```
czy to bezpieczne? ;)

tomeks86 commented on frontend/src/app/models/board/board-of-cells.ts:
```diff
@@ -0,0 +1,59 @@
+    this.board = [[]];
+  }
+
+  public isShipOnCell(y, x): boolean {
```
może jakieś coś znaczące nazwy współrzędnych?

tomeks86 commented on frontend/src/app/models/board/cell/cell.ts:
```diff
@@ -0,0 +1,23 @@
+    this.content = content;
+  }
+
+  setHereShip() {
```
putSheepHere?

tomeks86 commented on frontend/src/app/models/board/cell/cell.ts:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  isThereShip(): boolean {
+    return this.content.isShip();
```
isShipPresent

tomeks86 commented on frontend/src/app/models/board/cell/cell.ts:
```diff
@@ -0,0 +1,23 @@
+  }
+
+  isThereWater(): boolean {
+    return this.content.isWater();
```
isWaterPresent

tomeks86 commented on frontend/src/app/services/ship-generator/coordinate-generator/coordinate-on-surronding.ts:
```diff
@@ -0,0 +1,86 @@
+  }
+
+  //underscore generated automatically, so I leave it like that
+  private _upperY;
```
co to za podkreślniki? (nie rozumiem tajpskripta)

tomeks86 reviewed (dismissed):
proszę się odnieść do komentarzy

RafalJDev committed:
Still refactoring. Applying changes thanks to Tomek Seidler.
Also some formatting changes (more java like,
cause teammates are java based and that's why all classes are affected).

One major change in rest/post/ship-sender.
Changing method return statement cause it's not working od deployed
docker images, probably due to auto url prefix that is added to url.
(not the same as on localhost)

RafalJDev committed:
Adding console log to debug docker container on vps

RafalJDev committed:
Adding console log to debug docker container on vps

RafalJDev committed:
Recreating missing RowArray class

RafalJDev committed:
Preparing start game button.

totine commented on backend/src/main/java/pl/krkteam/controllers/ShipController.java:
```diff
@@ -0,0 +1,44 @@
+
+    ShipFromFronted shipFromFronted = gson.fromJson(post, ShipFromFronted.class);
+
+    System.out.println(shipFromFronted);
```
print

totine commented on backend/src/main/java/pl/krkteam/model/ShipFromFronted.java:
```diff
@@ -0,0 +1,15 @@
+
+import java.util.Arrays;
+
+public class ShipFromFronted {
```
DRY

gacikjakub reviewed (dismissed):
Nice object-oriented programming. :+1:

totine commented on frontend/src/app/services/ship-generator/ship-generator.service.ts:
```diff
@@ -0,0 +1,64 @@
+  generateShipsRandomly(board: BoardOfCells): ShipArray {
+    this.arrayOfMastToGenerate.forEach(mast => {
+      const mastCount = mast.count;
+      const isHorizontal = DirectionGenerator.isHorizontal();
```
Random in name

totine commented on frontend/src/app/services/ship-generator/ship-generator.service.ts:
```diff
@@ -0,0 +1,64 @@
+      const mastCount = mast.count;
+      const isHorizontal = DirectionGenerator.isHorizontal();
+  
+      let coordinate = this.coordinateGenerator.createHeadCoordinate(board, mastCount, isHorizontal);
```
Random in name

totine commented on frontend/src/app/services/ship-generator/coordinate-generator/coordinate-generator.service.ts:
```diff
@@ -0,0 +1,54 @@
+      coordinate = new Coordinate(rowIndex, columnIndex);
+    } while (this.isThereShipInSurrounding(coordinate, board, mastCount, isHorizontal))
+    
+    return new Coordinate(rowIndex, columnIndex);
```
Self comment: 2x new Coordinate

totine commented on frontend/src/styles.css:
```diff
@@ -0,0 +1,31 @@
+  margin-top: auto;
+}
+
+.ship1 {
```
ship1, ship2...

totine commented on frontend/src/index.html:
```diff
@@ -0,0 +1,16 @@
+      integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+<head>
+  <meta charset="utf-8">
+  <title>Ui</title>
```
change title to more descriptive

totine commented on frontend/src/app/views/game/start-game-button/start-game-button.component.ts:
```diff
@@ -0,0 +1,20 @@
+           })
+export class StartGameButtonComponent implements OnInit {
+  
+  //@Input() startGame = false;
```
to remove

totine commented on frontend/src/app/views/board/board.component.ts:
```diff
@@ -0,0 +1,71 @@
+  
+  showFieldBorder: boolean;
+  
+  //@Output()
```
to remove?

RafalJDev committed:
Refactoring: Generators to RandomGenerators.
Code formatting

RafalJDev committed:
Adding opponent board and player board.
